### ISSUE TITLE
affected/package: golang/dl Cannot install version go1.10 (or any other version) via go install [undefined: signalsToIgnore]

### ISSUE THREAD

**Issue by gavD** [golang/go#64115]
### What version of Go are you using (`go version`)?

<pre>
$ go version
go version go1.18.1 linux/amd64
</pre>

### Does this issue reproduce with the latest release?

Yes

### What operating system and processor architecture are you using (`go env`)?

Amazon Linux

<details><summary><code>go env</code> Output</summary><br><pre>

==> amazon-ebs: + go env
--
477 | amazon-ebs: Go env is:
478 | amazon-ebs: GO111MODULE=""
479 | amazon-ebs: GOARCH="amd64"
480 | amazon-ebs: GOBIN=""
481 | amazon-ebs: GOCACHE="/home/packer/.cache/go-build"
482 | amazon-ebs: GOENV="/home/packer/.config/go/env"
483 | amazon-ebs: GOEXE=""
484 | amazon-ebs: GOEXPERIMENT=""
485 | amazon-ebs: GOFLAGS=""
486 | amazon-ebs: GOHOSTARCH="amd64"
487 | amazon-ebs: GOHOSTOS="linux"
488 | amazon-ebs: GOINSECURE=""
489 | ==> amazon-ebs: + go install golang.org/dl/go1.10@latest
490 | amazon-ebs: GOMODCACHE="/home/packer/go/pkg/mod"
491 | amazon-ebs: GONOPROXY=""
492 | amazon-ebs: GONOSUMDB=""
493 | amazon-ebs: GOOS="linux"
494 | amazon-ebs: GOPATH="/home/packer/go"
495 | amazon-ebs: GOPRIVATE=""
496 | amazon-ebs: GOPROXY="https://proxy.golang.org,direct"
497 | amazon-ebs: GOROOT="/usr/lib/go-1.18"
498 | amazon-ebs: GOSUMDB="sum.golang.org"
499 | amazon-ebs: GOTMPDIR=""
500 | amazon-ebs: GOTOOLDIR="/usr/lib/go-1.18/pkg/tool/linux_amd64"
501 | amazon-ebs: GOVCS=""
502 | amazon-ebs: GOVERSION="go1.18.1"
503 | amazon-ebs: GCCGO="gccgo"
504 | amazon-ebs: GOAMD64="v1"
505 | amazon-ebs: AR="ar"
506 | amazon-ebs: CC="gcc"
507 | amazon-ebs: CXX="g++"
508 | amazon-ebs: CGO_ENABLED="1"
509 | amazon-ebs: GOMOD="/dev/null"
510 | amazon-ebs: GOWORK=""
511 | amazon-ebs: CGO_CFLAGS="-g -O2"
512 | amazon-ebs: CGO_CPPFLAGS=""
513 | amazon-ebs: CGO_CXXFLAGS="-g -O2"
514 | amazon-ebs: CGO_FFLAGS="-g -O2"
515 | amazon-ebs: CGO_LDFLAGS="-g -O2"
516 | amazon-ebs: PKG_CONFIG="pkg-config"
517 | amazon-ebs: GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build4109463442=/tmp/go-build -gno-record-gcc-switches"

</pre></details>

### What did you do?

We are building an AMI to run a legacy product.

```bash
apt-get install -y golang git

# fault is on the line below due to commit of the build of Go DL that seems to be breaking for us https://github.com/golang/dl/commit/8165dcd9305867c29bca9a55f2c45d3c12ccb1b7
go install golang.org/dl/go1.10@latest
```

1. We are building a legacy tool that requires Go 1.10.
2. Up until 6 days ago, this built fine but it has stopped working
3. We believe the issue is something like this:
    ![image](https://github.com/golang/go/assets/205245/1618d399-9e7c-4ef0-8c68-41f1de60d125)
4. We suspect the causal change may be related to this: https://github.com/golang/dl/commit/8165dcd9305867c29bca9a55f2c45d3c12ccb1b7
5. We tried instead installing go 1.20 and got the same issue

### What did you expect to see?

Prior to 6 days ago, our legacy product built correctly.

### What did you see instead?

```
==> amazon-ebs: + go install golang.org/dl/go1.10@latest
--
470 | ==> amazon-ebs: go: downloading golang.org/dl v0.0.0-20231107201224-8165dcd93058
471 | ==> amazon-ebs: # golang.org/dl/internal/version
472 | ==> amazon-ebs: go/pkg/mod/golang.org/dl@v0.0.0-20231107201224-8165dcd93058/internal/version/version.go:537:38: undefined: signalsToIgnore
```

**Comment by seankhliao** [comment]
1.18 is outside of our support window, please use a newer version to install the go1.10 wrapper

**Comment by gavD** [comment]
thank you Sean, it is building now we've upgraded the system version of Go

**Comment by njelich** [comment]
Could this error be improved to make it clear the local go version is outdated.

**Comment by eldy** [comment]
Same trouble. Solved by installing go 1.20 manually with the method 2 of this tutorial:
https://tecadmin.net/how-to-install-go-on-ubuntu-20-04/
